util/cache/redis.go

```diff
@@ -5,59 +5,52 @@ import (
 	"time"
 
 	ioutil "github.com/argoproj/gitops-engine/pkg/utils/io"
-	rediscache "github.com/go-redis/cache"
-	"github.com/go-redis/redis"
-	"github.com/vmihailenco/msgpack"
+	rediscache "github.com/go-redis/cache/v8"
+	"github.com/go-redis/redis/v8"
 )
 
 func NewRedisCache(client *redis.Client, expiration time.Duration) CacheClient {
 	return &redisCache{
 		client:     client,
 		expiration: expiration,
-		codec: &rediscache.Codec{
-			Redis: client,
-			Marshal: func(v interface{}) ([]byte, error) {
-				return msgpack.Marshal(v)
-			},
-			Unmarshal: func(b []byte, v interface{}) error {
-				return msgpack.Unmarshal(b, v)
-			},
-		},
+		cache:      rediscache.New(&rediscache.Options{Redis: client}),
 	}
 }
 
 type redisCache struct {
 	expiration time.Duration
 	client     *redis.Client
-	codec      *rediscache.Codec
+	cache      *rediscache.Cache
 }
 
 func (r *redisCache) Set(item *Item) error {
 	expiration := item.Expiration
 	if expiration == 0 {
 		expiration = r.expiration
 	}
-	return r.codec.Set(&rediscache.Item{
-		Key:        item.Key,
-		Object:     item.Object,
-		Expiration: expiration,
+
+	return r.cache.Set(&rediscache.Item{
+		Key:   item.Key,
+		Value: item.Object,
+		TTL:   expiration,
 	})
 }
 
 func (r *redisCache) Get(key string, obj interface{}) error {
-	err := r.codec.Get(key, obj)
+	err := r.cache.Get(context.TODO(), key, obj)
```

Comment:
We might want to propagate contexts to these functions in the future if available, however I used `context.TODO` to keep this PR focused/minimal